operator/src/main/java/oracle/kubernetes/operator/steps/ManagedServersUpStep.java

```diff
@@ -74,18 +74,28 @@ private static Step scaleDownIfNecessary(
     List<ServerShutdownInfo> serversToStop = getServersToStop(info, factory.shutdownInfos);
 
     if (!serversToStop.isEmpty()) {
-      insert(steps,
-          Step.chain(createProgressingStartedEventStep(info, MANAGED_SERVERS_STARTING_PROGRESS_REASON, true,
-          null), new ServerDownIteratorStep(factory.shutdownInfos, null)));
+      if (isServiceOnlyServers(serversToStop)) {
```

Comment:
I am a little confused about the logic here. If the server pod is not null, the pod will be taken down no matter if the server is serviceOnly or not. If there are servers (pod or/and service) to be shutdown, the domain's progressing status needs to be updated, and the DomainProcessingStarting event should be generated. Only if there is no pod/service that needs to be taken down, the domain status update and event generation can be skipped.

Reply:
The service is deleted only if `isPreserveServices` option is not set to `true`. I believe when `precreateServerServices` is `true`, `isPreserveServices` option is also set to `true`. So in this case, the service will not be deleted and the whole step will be a `noop`. When `precreateServerServices` is `true` and service doesn't exist (which will be the case during first time), this step will create the service. Please see my comment above - https://github.com/oracle/weblogic-kubernetes-operator/pull/2124#discussion_r552255595

Reply:
I think we need to update the domain status and generate the event unless ServerDownIteratorStep is a noop.

Reply:
Latest changes in c26deeb will generate the event unless the ServerDownIteratorStep is a noop. Thanks.